nilearn/masking.py

```diff
@@ -607,32 +613,47 @@ def compute_brain_mask(target_img, threshold=.5, connected=True,
         Controls the amount of verbosity: higher numbers give
         more messages
 
+    mask_type : {'whole-brain', 'gm', 'wm'}, optional
+        Type of mask to be computed. Default = 'whole-brain'
+
+    resolution: int, optional, Default = 2
+        If resolution is different from 1, the original template is re-sampled
+        with the specified resolution and a mask with the same resolution is
+        computed.
+
     Returns
     -------
     mask : nibabel.Nifti1Image
         The whole-brain mask (3D image)
     """
     if verbose > 0:
-        print("Template mask computation")
+        msg_to_print = "Template " + mask_type + " mask computation"
+        print(msg_to_print)
 
     target_img = _utils.check_niimg(target_img)
 
-    from .datasets import load_mni152_brain_mask
-    template = load_mni152_brain_mask()
+    if mask_type == 'whole-brain':
+        template = load_mni152_template(resolution=resolution)
+    elif mask_type == 'gm':
+        template = load_mni152_gm_template(resolution=resolution)
+    elif mask_type == 'wm':
+        template = load_mni152_wm_template(resolution=resolution)
+    else:
+        raise ValueError("Unknown mask type {}.".format(mask_type))
+
     dtype = img_data_dtype(target_img)
-    template = new_img_like(template,
-                            get_data(template).astype(dtype))
+    template = new_img_like(template, get_data(template).astype(dtype))
 
-    from .image.resampling import resample_to_img
-    resampled_template = cache(resample_to_img, memory)(template, target_img)
+    resampled_template = cache(resampling.resample_to_img, memory)(
```

Comment:
why do we need `force_resample=True` ? according to the docstring of `resample_to_img` it is "intended for testing"

Reply:
We decided to force the resampling at some point in this PR, see https://github.com/nilearn/nilearn/pull/2738#issuecomment-810024868

I think that removing this is currently causing some issues (at least there is one test failing...). I'll have a closer look if you don't think we should rely on `force_resample`. I suspect the issue might also come from the failing test itself... :thinking:

Reply:
it could also be a bug in `resample_img`, because from the description it sounds like `force_resample` should make the function a bit slower but shouldn't change the output. In any case this PR's page is already huge so we can investigate this separately; and maybe just add a small comment explaining why `force_resample` is needed?

Reply:
@jeromedockes Yes I agree that we might want to have a closer look in a separate issue/PR as this one is gettting quite large. I'll try to come up with a small reproducing example.

Reply:
I think this has been resolved